#! /bin/sh
patch -p0 -f $* < $0
exit $?

Use binary open mode for files. 
Fix keyserver list.


2006-02-15  Werner Koch  <wk@g10code.com>

       * Fixed typo in keyserver list.

2006-02-14  Marcus Brinkmann  <marcus@g10code.de>

       * gpgmetools.c (O_BINARY): Define if it isn't defined by system
       headers.
       (gpa_open_output): Use open() instead of creat() and set binary
       mode.
       (gpa_open_input): Set binary mode.



Index: src/gpgmetools.c
===================================================================
--- src/gpgmetools.c    (revision 738)
+++ src/gpgmetools.c    (working copy)
@@ -43,6 +43,12 @@
 #include <windows.h>
 #endif
 
+#ifndef O_BINARY
+#ifdef _O_BINARY
+#define O_BINARY       _O_BINARY
+#else
+#define O_BINARY       0
+#endif
 
 /* Report an unexpected error in GPGME and quit the application.  */
 void
@@ -176,7 +182,7 @@
     {
       gpg_error_t err;
 
-      target = creat (filename, 0666);
+      target = open (filename, O_WRONLY | O_CREAT | O_TRUNC | O_BINARY, 0666);
       if (target == -1)
        {
          gchar *message;
@@ -202,7 +208,7 @@
   gpg_error_t err;
   int target = -1;
 
-  target = open (filename, O_RDONLY);
+  target = open (filename, O_RDONLY | O_BINARY);
   if (target == -1)
     {
       gchar *message;
Index: src/keyserver.c
===================================================================
--- src/keyserver.c     (revision 739)
+++ src/keyserver.c     (working copy)
@@ -169,7 +169,7 @@
       add_server (&serverlist, "hkp://random.sks.keyserver.penguin.de");
       add_server (&serverlist, "hkp://blackhole.pca.dfn.de"); 
       add_server (&serverlist, "hkp://pks.gpg.cz");
-      add_server (&serverlist, "hpk://pgp.cns.ualberta.ca");
+      add_server (&serverlist, "hkp://pgp.cns.ualberta.ca");
       add_server (&serverlist, "hkp://minsky.surfnet.nl");
       add_server (&serverlist, "hkp://keyserver.ubuntu.com");
       add_server (&serverlist, "hkp://keyserver.pramberger.at");

